src/main/java/bio/terra/service/snapshotbuilder/query/Query.java

```diff
@@ -108,11 +115,17 @@ public String renderSQL() {
       sql += " " + having.renderSQL();
     }
 
-    // TODO: DC-836 Implement LIMIT for Azure (TOP N + sql)
-    //  This means passing in the platform to renderSQL
-    //  and refactoring the current TableVariable and tableNameGenerator work
     if (limit != null) {
-      sql += " LIMIT " + limit;
+      if (platform != null) {
+        if (platform.isGcp()) {
+          sql += " LIMIT " + limit;
+        } else if (platform().isAzure()) {
+          sql = "TOP " + limit + " " + sql;
+        }
+        // if platform is not provided limit for GCP is generated
```

Comment:
Once you have the check in the constructor, you can remove this code. If you did want to use GCP as the default, you could add a constructor that omits `platform` and provides `GCP` instead of `null`. But in our use cases the platform should always be set.